### PR TITLE
add slope == 0 case into standard leaky relu nn test

### PR DESCRIPTION
**Pull request by glaringlee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37559 add slope == 0 case into standard leaky relu nn test**

Differential Revision: [D21319922](https://our.internmc.facebook.com/intern/diff/D21319922)

### CODE REVIEW

File: test/test_autograd.py [hunk 1]
```diff
@@ -5635,11 +5635,6 @@ def test_leaky_relu_inplace_with_zero_or_neg_slope(self, device):
         with self.assertRaisesRegex(RuntimeError, "call out-of-place version"):
             b.backward(torch.ones(2, device=device))
 
-        a = torch.tensor([-2., 0., 2.], device=device, requires_grad=True)
```
Comment (albanD) [inline test/test_autograd.py:5638]: I think this is worth keeping!
We do rely on this subgradient being 0. Even though we most likely will never willingly change that. It is good to have a test that makes sure it does not happen by mistake.

**Review by albanD — CHANGES_REQUESTED** [review]
Thanks for the PR.
The new test looks good !

File: test/test_autograd.py (hunk 1 above)
Comment (glaringlee) [reply test/test_autograd.py:5638]: sure. I will make a separate test to test this.

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**